server/sample.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/prometheus/prometheus/scrape"
+	"golang.org/x/exp/slices"
+)
+
+func mutateSampleLabels(lset labels.Labels, target *scrape.Target, honor bool, rc []*relabel.Config) labels.Labels {
+	lb := labels.NewBuilder(lset)
+
+	if honor {
```

Comment:
Let's rename to `honorTs`. Without that, it's hard to read.

Also, I don't understand this part. What does honor timestamp mean? Who will set this? And why would there be a conflict?

Reply:
From what I understand, it's whether the scrape target includes timestamps for the samples. You can add timestamp for openmetrics data and in that case you can choose to use that instead of scrape time, it's an optional case though.

Reply:
In our case, the target is always prometheus right? Can we just hardcode to that case?

Reply:
Sorry the target is mattermost, but yes we never include timestamps so we can actually hardcode for our case.

Reply:
Oh yeah sorry. Cool sounds good then :+1: